src/main/java/io/cryostat/rules/PeriodicArchiver.java

```diff
@@ -105,50 +98,30 @@ public void run() {
         } catch (InterruptedException | ExecutionException e) {
             logger.error(e);
             failureNotifier.apply(Pair.of(serviceRef, rule));
+        } catch (Exception e) {
+            throw new HttpStatusException(500, e);
         }
     }
 
-    void performArchival() throws InterruptedException, ExecutionException {
-        // FIXME using an HTTP request to localhost here works well enough, but is needlessly
-        // complex. The API handler targeted here should be refactored to extract the logic that
-        // creates the recording from the logic that simply figures out the recording parameters
-        // from the POST form, path param, and headers. Then the handler should consume the API
-        // exposed by this refactored chunk, and this refactored chunk can also be consumed here
-        // rather than firing HTTP requests to ourselves
+    void performArchival() throws InterruptedException, ExecutionException, Exception {
 
+        String recordingName = rule.getRecordingName();
+        ConnectionDescriptor connectionDescriptor =
+                new ConnectionDescriptor(serviceRef, credentialsManager.getCredentials(serviceRef));
         String path =
```

Comment:
Do we still need this path?

Reply:
The path is only used for the logger message above. Since we're moving the log message, `path` can be removed too